osu.Game/Beatmaps/ControlPoints/ControlPoint.cs

```diff
@@ -14,6 +14,16 @@ public class ControlPoint : IComparable<ControlPoint>, IEquatable<ControlPoint>
 
         public int CompareTo(ControlPoint other) => Time.CompareTo(other.Time);
 
-        public bool Equals(ControlPoint other) => Time.Equals(other?.Time);
+        /// <summary>
+        /// Whether this <see cref="ControlPoint"/> provides the same changes to gameplay as another <see cref="ControlPoint"/>.
+        /// </summary>
+        /// <param name="other">The <see cref="ControlPoint"/> to compare to.</param>
+        /// <returns>Whether this <see cref="ControlPoint"/> provides the same changes to gameplay as <paramref name="other"/>.</returns>
+        public virtual bool ChangeEquals(ControlPoint other) => !ReferenceEquals(null, other);
```

Comment:
Not sure I like this method name.. how about `HasParameterChanges` or something?

Reply:
also not sure there should be a base implementation (unsure of the value of ReferenceEquals / null check here vs always true).